scripts/install_prerequesites_ubuntu_24_04.sh

```diff
@@ -1,20 +1,45 @@
 #!/bin/bash
 
-# Add Docker's official GPG key:
-sudo apt-get update -y
-sudo apt-get install ca-certificates curl -y
-sudo install -m 0755 -d /etc/apt/keyrings
-sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
-sudo chmod a+r /etc/apt/keyrings/docker.asc
+PIPENV_CMD="pipenv"
 
-# Add the docker repository to Apt sources:
-echo \
-    "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
-  $(. /etc/os-release && echo "${UBUNTU_CODENAME:-$VERSION_CODENAME}") stable" |
-    sudo tee /etc/apt/sources.list.d/docker.list >/dev/null
-sudo apt-get update -y
+install_docker() {
+    # Add Docker's official GPG key:
```

Comment:
please add a check to see if docker is installed and if so skip the rest of the step